packages/cli/src/runners/task-managers/data-request-response-builder.ts

```diff
@@ -0,0 +1,199 @@
+			node: taskData.node, // The current node being executed
+			runExecutionData: this.buildRunExecutionData(taskData.runExecutionData),
+			runIndex: taskData.runIndex,
+			selfData: taskData.selfData,
+			siblingParameters: taskData.siblingParameters,
+			executeData: this.buildExecuteData(taskData.executeData),
+			additionalData: this.buildAdditionalData(taskData.additionalData),
+		};
+	}
+
+	buildAdditionalData(additionalData: IWorkflowExecuteAdditionalData): PartialAdditionalData {
+		return {
+			formWaitingBaseUrl: additionalData.formWaitingBaseUrl,
+			instanceBaseUrl: additionalData.instanceBaseUrl,
+			restApiUrl: additionalData.restApiUrl,
+			variables: additionalData.variables,
+			webhookBaseUrl: additionalData.webhookBaseUrl,
+			webhookTestBaseUrl: additionalData.webhookTestBaseUrl,
+			webhookWaitingBaseUrl: additionalData.webhookWaitingBaseUrl,
+			currentNodeParameters: additionalData.currentNodeParameters,
+			executionId: additionalData.executionId,
+			executionTimeoutTimestamp: additionalData.executionTimeoutTimestamp,
+			restartExecutionId: additionalData.restartExecutionId,
+			userId: additionalData.userId,
+		};
+	}
+
+	buildExecuteData(executeData: IExecuteData | undefined): IExecuteData | undefined {
+		if (executeData === undefined) {
+			return undefined;
+		}
+
+		return {
+			node: executeData.node, // The current node being executed
+			data: this.requestParams.input ? executeData.data : {},
+			source: executeData.source,
+		};
+	}
+
+	buildRunExecutionData(runExecutionData: IRunExecutionData): IRunExecutionData {
+		if (this.requestParams.dataOfNodes === 'all') {
+			return runExecutionData;
+		}
+
+		return {
+			startData: runExecutionData.startData,
+			resultData: {
+				error: runExecutionData.resultData.error,
+				lastNodeExecuted: runExecutionData.resultData.lastNodeExecuted,
+				metadata: runExecutionData.resultData.metadata,
+				runData: this.buildRunData(runExecutionData.resultData.runData),
+				pinData: this.buildPinData(runExecutionData.resultData.pinData),
+			},
+			executionData: runExecutionData.executionData
+				? {
+						// TODO: Figure out what these two are and can they be filtered
+						contextData: runExecutionData.executionData?.contextData,
+						nodeExecutionStack: runExecutionData.executionData.nodeExecutionStack,
+
+						metadata: runExecutionData.executionData.metadata,
+						waitingExecution: runExecutionData.executionData.waitingExecution,
+						waitingExecutionSource: runExecutionData.executionData.waitingExecutionSource,
+					}
+				: undefined,
+		};
+	}
+
+	buildRunData(runData: IRunData): IRunData {
+		return this.filterObjectByNodeNames(runData);
+	}
+
+	buildPinData(pinData: IPinData | undefined): IPinData | undefined {
+		return pinData ? this.filterObjectByNodeNames(pinData) : undefined;
+	}
+
+	buildEnvProviderState(envProviderState: EnvProviderState): EnvProviderState {
+		if (this.requestParams.env) {
+			return envProviderState;
+		}
```

Comment:
Don't we want to provide it only if `!envProviderState.isEnvAccessBlocked`?

Reply:
Good question. Those have already been handled where the state is created. Added a clarifying comment